runtime/include/atomics/standard/chpl-atomics.h

```diff
@@ -0,0 +1,295 @@
+} \
+static inline type atomic_exchange_ ## type(atomic_ ## type * obj, type value) { \
+  return atomic_exchange(obj, value); \
+} \
+static inline chpl_bool atomic_compare_exchange_strong_explicit_ ## type(atomic_ ## type * obj, type expected, type desired, memory_order order) { \
+  return atomic_compare_exchange_strong_explicit(obj, &expected, desired, order, order); \
+} \
+static inline chpl_bool atomic_compare_exchange_strong_ ## type(atomic_ ## type * obj, type expected, type desired) { \
+  return atomic_compare_exchange_strong(obj, &expected, desired); \
+} \
+static inline chpl_bool atomic_compare_exchange_weak_explicit_ ## type(atomic_ ## type * obj, type expected, type desired, memory_order order) { \
+  return atomic_compare_exchange_weak_explicit(obj, &expected, desired, order, order); \
+} \
+static inline chpl_bool atomic_compare_exchange_weak_ ## type(atomic_ ## type * obj, type expected, type desired) { \
+  return atomic_compare_exchange_weak(obj, &expected, desired); \
+}
+
+
+///////////////////////////////////////////////////////////////////////////////
+////                   START OF REAL ATOMICS FETCH OPS                    ////
+//////////////////////////////////////////////////////////////////////////////
+#define DECLARE_REAL_ATOMICS_FETCH_OPS(type) \
+static inline type atomic_fetch_add_explicit_ ## type(atomic_ ## type * obj, type operand, memory_order order) { \
+  type old_val = *obj; \
+  type new_val; \
+  do { \
+    new_val = old_val + operand; \
+  } while (!atomic_compare_exchange_weak_explicit(obj, &old_val, new_val, order, order)); \
+  return old_val; \
+} \
+static inline type atomic_fetch_add_ ## type(atomic_ ## type * obj, type operand) { \
+  return atomic_fetch_add_explicit_ ## type(obj, operand, memory_order_seq_cst); \
+} \
+static inline type atomic_fetch_sub_explicit_ ## type(atomic_ ## type * obj, type operand, memory_order order) { \
+  type old_val = *obj; \
+  type new_val; \
+  do { \
+    new_val = old_val - operand; \
+  } while (!atomic_compare_exchange_weak_explicit(obj, &old_val, new_val, order, order)); \
+  return old_val; \
+} \
+static inline type atomic_fetch_sub_ ## type(atomic_ ## type * obj, type operand) { \
+  return atomic_fetch_sub_explicit_ ## type(obj, operand, memory_order_seq_cst); \
+}
+
+
+// Actually declare the atomics for integer and real types using the above macros
+DECLARE_ATOMICS_BASE(bool, chpl_bool);
+DECLARE_ATOMICS_EXCHANGE_OPS(bool, chpl_bool);
+
+
+#define DECLARE_ATOMICS(type) \
+  DECLARE_ATOMICS_BASE(type, type) \
+  DECLARE_ATOMICS_EXCHANGE_OPS(type, type); \
+  DECLARE_ATOMICS_FETCH_OPS(type)
+
+DECLARE_ATOMICS(int_least8_t);
+DECLARE_ATOMICS(int_least16_t);
+DECLARE_ATOMICS(int_least32_t);
+DECLARE_ATOMICS(int_least64_t);
+DECLARE_ATOMICS(uint_least8_t);
+DECLARE_ATOMICS(uint_least16_t);
+DECLARE_ATOMICS(uint_least32_t);
+DECLARE_ATOMICS(uint_least64_t);
+
+// On netbsd the DECLARE_ATOMICS macro doesn't work for uintptr_t. From gbt:
```

Comment:
I know that I wrote this comment and you just copied it from the intrinsics version, but it's really long :) I think it'd be better in here to just refer to the comment in the intrinsics backend, or refer to the sha of the git commit for history instead of duplicating the comment here